spec.bs

```diff
@@ -1306,20 +1310,27 @@ maps to an internal [=fenced frame config=] [=struct=].
   };
 </pre>
 
-Note: Note that {{FencedFrameConfig}}s cannot be constructed manually from JavaScript. They can only
-be created by config-generating APIs. However, some browsers [support a developer-only
-flag](https://github.com/WICG/fenced-frame/issues/94) which help developers test the <{fencedframe}>
-element with arbitrary URLs *not* produced by config-generating APIs for development.
-
 Each {{FencedFrameConfig}} has:
 
+ * A <dfn for=fencedframeconfig>url</dfn>, a [=URL=] or null, initially null
  * A <dfn for=fencedframeconfig>urn</dfn>, a [=urn uuid=]
  * A <dfn for=fencedframeconfig>sharedStorageContext</dfn>, a [=string=]
  * A <dfn for=fencedframeconfig>containerWidth</dfn>, a {{FencedFrameConfigSize}} or null
  * A <dfn for=fencedframeconfig>containerHeight</dfn>, a {{FencedFrameConfigSize}} or null
  * A <dfn for=fencedframeconfig>contentWidth</dfn>, a {{FencedFrameConfigSize}} or null
  * A <dfn for=fencedframeconfig>contentHeight</dfn>, a {{FencedFrameConfigSize}} or null
 
+<div algorithm>
+  The <dfn constructor for=FencedFrameConfig>FencedFrameConfig(|url|)</dfn> constructor method steps
+  are:
+  
+  1. Let |config| be a newly constructed {{FencedFrameConfig}} object.
+
+  2. Set |config|'s [=fencedframeconfig/url=] to the result of running the [=URL parser=] on |url|.
```

Comment:
So the parser returns either a URL or "failure", I believe. It doesn't seem to return null at all, so blindly setting this is probably wrong. I think it's best to maybe make the member _also_ a URL or failure, and then making the navigation steps simply fail if the FFC/url member is "failure". Does that make sense?

Reply:
We'd need to distinguish between the failure case (i.e. a URL was attempted to be set but something went wrong) and the default case (the URL hasn't been set yet, and this config might have a URN instead that we need to use). I think we can have `fencedframeconfig/url` be `a URL, failure, or null, initially null` (so that `null` is still used for the unset case), and then `failure` will just be for the failure case.